components/core/include/core/util/hash.hpp

```diff
@@ -38,4 +43,11 @@ inline void hash_combine(size_t &seed, const T &v)
 
 	hash_combine(seed, hasher(v));
 }
+
+inline std::string sha256(const std::string &input)
```

Comment:
Might want to consider using a `std::array<uint8_t, 32>`  (or `std::array<char, 64>` if you want to keep it hex encoded) for hashes since `std::array` can be `constexpr` and doesn't ever dynamically allocate.